### PR TITLE
Create 15SP4 qcow for SLEHPC via autoyast

### PR DESCRIPTION
**Pull request by lemon-suse**
We need create a daily auto-installation for SLE HPC 15 SP4 with necessary specifics, include qcows for 3 roles( ld - HPC Login and Development Node, ms - HPC Management Server, tm - text mode).

- Related ticket: https://progress.opensuse.org/issues/120870
- Needles: N/A
- Related MR: https://gitlab.suse.de/qa-maintenance/qam-openqa-yml/-/merge_requests/443
                         https://gitlab.suse.de/coolgw/wegao-test/-/merge_requests/208
- Verification run: 
  Auto-installation: https://openqa.nue.suse.com/tests/overview?distri=sle&version=15-SP4&build=20230110-1&groupid=316
  X86_64:  
  migration VRs:
  http://openqa.nue.suse.com/tests/10320143#  textmode, all patterns
  http://openqa.nue.suse.com/tests/10320142#  textmode, defaut patterns
  http://openqa.nue.suse.com/tests/10320145#   management server role, default patterns
  http://openqa.nue.suse.com/tests/10320146#  management server role, all patterns
  http://openqa.nue.suse.com/tests/10320147#  development node role, default patterns
  http://openqa.nue.suse.com/tests/10320148#  development node role, all patterns

  aarch64: 
   migration VRs:
   http://openqa.nue.suse.com/tests/10320149#       textmode,default
   http://openqa.nue.suse.com/tests/10320150#      textmode, all patterns
   http://openqa.nue.suse.com/tests/10320152#     management server role, default patterns 
  http://openqa.nue.suse.com/tests/10320151#     management server role, all patterns
  http://openqa.nue.suse.com/tests/10320154#     development node role, default patterns  
  http://openqa.nue.suse.com/tests/10320153#    development node role, all patterns

### CODE REVIEW

File: data/autoyast_sle15/create_hdd/create_hdd_hpc_development_node.xml.ep [hunk 1]
```diff
@@ -0,0 +1,314 @@
...
+    <packages config:type="list">
+      <!-- Workaround for bsc#1202234: [addon]-release packages are missing after autoyast installation. -->
+      % foreach (values %$addons) {
+      <package><%= lc($_->{name}) %>-release</package>
+      % }
+      <!-- end of workaround -->
+    </packages>
+    % if ($check_var->('PATTERNS', 'all')) {
+    <patterns config:type="list">
+      <pattern>32bit</pattern>
```
Comment (JRivrain) [inline data/autoyast_sle15/create_hdd/create_hdd_hpc_development_node.xml.ep:194]: If I compare with the ticket, those are missing:
< common-criteria
< devel_basis
< devel_kernel
< devel_yast

**Comment by JRivrain** [comment]
What did you use as a base? It's recommended to use this method to create the base profile https://confluence.suse.com/display/qasle/How-To%3A+Adapt+an+openQA+installation+testsuite+to+use+AutoYaST#HowTo:AdaptanopenQAinstallationtestsuitetouseAutoYaST-Acquirethe(compact)autoyastprofileofcreate_hdd_gnome 
Then you can adapt, it avoids mistakes and deprecated options. It does not work on sle12 but is the preferred method for sle15.

**Review by ge0r — APPROVED** [review]

File: data/autoyast_sle15/create_hdd/create_hdd_hpc_development_node_x86_64.xml.ep [hunk 2]
```diff
@@ -0,0 +1,475 @@
...
+    <install_recommended t="boolean">true</install_recommended>
+    <instsource/>
+    <packages config:type="list">
+      <!-- Workaround for bsc#1202234: [addon]-release packages are missing after autoyast installation. -->
+      % foreach (values %$addons) {
+      <package><%= lc($_->{name}) %>-release</package>
+      % }
+      <!-- end of workaround -->
+    </packages>
+    % if ($check_var->('PATTERNS', 'all')) {
```
Comment (JRivrain) [inline data/autoyast_sle15/create_hdd/create_hdd_hpc_development_node_x86_64.xml.ep:312]: I see you added many new patterns that are no in the list in the ticket, why is that ? for example with development node, all patterns:
jonathan@linux-hh22:~/blouk> diff dev-text-all.txt dev-xml-all
2a3
> apparmor-32bit
3a5,7
> base-32bit
> basesystem
> basic_desktop
11a16
> enhanced_base-32bit
16a22
> gnome_basis
24a31,32
> minimal_base
> minimal_base-32bit
29a38,41
> sw_management-32bit
> wsl_base
> wsl_gui
> wsl_systemd
30a43,45
> x11-32bit
> x11_enhanced
> x11_enhanced-32bit

File: tests/autoyast/clone.pm [hunk 3]
```diff
@@ -19,7 +19,8 @@ sub run {
     my $self = shift;
     assert_script_run 'rm -f /root/autoinst.xml';
     zypper_call('in autoyast2', 300);
-    my $module_name = y2_module_consoletest::yast2_console_exec(yast2_module => 'clone_system', yast2_opts => '--ncurses');
+    #my $module_name = y2_module_consoletest::yast2_console_exec(yast2_module => 'clone_system', yast2_opts => '--ncurses');
```
[the remark this answers is not in the document]
Comment (lemon-suse) [reply tests/autoyast/clone.pm:22]: Ok, I will remove it when got all cloned profiles. Thanks.

**Review by JRivrain — APPROVED** [review]

**Pull request merged**